File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/managetaskview/TaskViewEvent.java
package com.wiseassblog.samsaradayplanner.ui.managetaskview;

/**
 * It models interactions between the View and the Logic class
 */
public class TaskViewEvent {

    private final Event event;
    private final Object value;

    public TaskViewEvent(Event event, Object value){
        this.event = event;
        this.value = value;
    }

    public Event getEvent() {
        return event;
    }

    public Object getValue() {
        return value;
    }

    /**
     * NOTE: ON_START expects a taskID to be sent in as the value
     */
    public enum Event {
        ON_COLOR_BUTTON_CLICK,
        ON_COLOR_SELECTED,
        ON_DONE_CLICK,
        ON_ICON_SELECTED,
        ON_START
    }
}

File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/managetaskview/TaskViewEvent.kt
package com.wiseassblog.samsaradayplanner.ui.managetaskview

import com.wiseassblog.samsaradayplanner.domain.constants.COLOR
import com.wiseassblog.samsaradayplanner.domain.constants.ICON

/**
 * It models interactions between the View and the Logic class
 */
sealed class TaskViewEvent {
    object OnColorButtonClick : TaskViewEvent()
    data class OnColorSelected(val color: COLOR) : TaskViewEvent()
    object OnDoneClick : TaskViewEvent()
    data class OnIconSelected(val icon: ICON) : TaskViewEvent()
    object OnStart : TaskViewEvent()
}